visualizer/actionButton.js

```diff
@@ -20,15 +24,146 @@
     LEFTWARDS_ARROW: '\u2190'
   };
 
-  function createSemanticNameContainer() {
+  var deleteEntry = document.querySelector('#action.contextMenu #delete');
+  var editEntry = document.querySelector('#action.contextMenu #edit');
+
+  // Unselect all the semantics buttons, except the target semantic button
+  function unselectOtherSemanticButtons(targetNameContainer) {
+    var wrappers = document.querySelectorAll('#semantics .wrapper');
+    Array.prototype.forEach.call(wrappers, function(wrapper) {
+      var nameContainer = wrapper.querySelector('textarea.opName');
+      if (targetNameContainer === nameContainer) {
+        return;
+      }
+      wrapper.classList.remove('selected');
+
+      // Hide the argument list if there is one
+      var arrowButton = wrapper.querySelector('.arrow');
+      if (arrowButton) {
+        arrowButton.innerHTML = UnicodeChars.BLACK_DOWN_POINTING_TRIANGLE;
+        wrapper.querySelector('.arguments').hidden = true;
+      }
+    });
+  }
+
+  function retrieveArgs(wrapper) {
+    var argList = wrapper.querySelector('.arguments');
+    var nameValPairs = wrapper.querySelectorAll('.nameValuePair');
+    var args = Object.create(null);
+    Array.prototype.forEach.call(nameValPairs, function(nameValPair) {
+      var argNameContainer = nameValPair.querySelector('.name');
+      var argName = argNameContainer.value;
+
+      var valueContainer = nameValPair.querySelector('.value');
+      var value = valueContainer.value;
+      if (argName && /^[_a-zA-Z0-9]+$/.test(argName) && !(argName in args)) {
```

Comment:
What's this regex doing? In the very least this could use a comment. It looks like it's testing for valid JavaScript identifiers, but we should be very careful doing that if the user has specified the name of the identifier. The other case was a bit different, because we are _generating_ the identifier name, so I think it's ok to restrict that to ASCII characters. But it looks like this is taking a name that comes from the user, and testing if it looks like a valid identifier -- is that right?

If that's the case, we should at least add a TODO here that says we should fix this to allow any valid JavaScript identifier, not just the restricted set that will match this regex.